gms/gossiper.cc

```diff
@@ -2292,7 +2292,7 @@ future<> gossiper::add_local_application_state(application_state_map states) {
             co_await gossiper.do_on_change_notifications(ep_addr, states, permit.id());
         });
     } catch (...) {
-        logger.warn("Fail to apply application_state: {}", std::current_exception());
+        logger.error("Fail to apply application_state: {}", std::current_exception());
```

Comment:
I wouldn't do this here in this PR, I expect to have it much larger consequences -- dtest failures and so on.

It's a good initiative, but it won't be so easy.
So please take this out, and send a separate PR, and run dtest-full job on that other PR so we can see what else will fail

Reply:
Also we might find that some error types deserve WARN while other deserve ERROR.

Could this failure happen e.g. due to transient network connectivity problem, and it will recover eventually? In that case we should probably use WARN. Etc. (that's just an example, looking at the code I don't see what kind of transient error could cause something like that, maybe something in `do_on_change_notifications`)

Reply:
Ok, removed the commit for now, keeping just the fix.

Reply:
It turns out
```
gossip - Fail to apply application_state: std::runtime_error (endpoint_state_map does not contain endpoint = 127.39.252.1, application_states = {CDC_GENERATION_ID
```
is logged on every restart so we for sure can't change the log level until we fix it.

Reply:
I initially thought that this could happen only during upgrade.